### DISCUSSION TITLE
Next.js \ Three.js / GLTFLoader

### DISCUSSION THREAD

**Discussion by enkicoma** [vercel/next.js#20244] [Help]
Hi guys
can you please help me to debug this issue or at least to explain where is the problem.. https://codesandbox.io/s/clever-worker-hd1bf?file=/components/Duck.js

```js
Server Error

SyntaxError: Unexpected token {
This error happened while generating the page. Any console logs will be displayed in the terminal window.
```

I like Next.js but I may switch back to normal react because too many module issues and too many things behind the scenes... 

Thanks

**Comment by JustFly1984** [comment]
@enkicoma You should not render canvas components on server side. You need to detect for typeof window !== 'undefined' before rendering any of Three.js components. Render the fallback instead for SSR or SSG.
